west.yml

```diff
@@ -0,0 +1,34 @@
+# SOF west manifest
+manifest:
+  defaults:
+    remote: thesofproject
+
+  remotes:
+    - name: thesofproject
+      url-base: https://github.com/thesofproject
+    - name: zephyrproject
+      url-base: https://github.com/zephyrproject-rtos
+
+  projects:
+    - name: rimage
+      repo-path: rimage
+      path: sof/rimage
+      revision: 02abc5d342a3ee6965bdc933ea1439d85d0256da
+      remote: thesofproject
+
+    - name: tomlc99
+      repo-path: tomlc99
+      path: sof/rimage/tomlc99
+      revision: e3a03f5ec7d8d33be705c5ce8a632d998ce9b4d1
+      remote: thesofproject
+
+    - name: zephyr
+      repo-path: zephyr
+      revision: 63a497280f22aea39793dd9de9731469a83ce533
+      remote: zephyrproject
+      import:
+        #remove this once SOF is removed from Zephyr manifest
+        path-blacklist: modules/audio/sof
```

Comment:
instead of blacklisting, you should be whitelisting I guess and only add the things you need. Start with only those:

          - hal_xtensa
          - mbedtls
          - mipi-sys-t

And then try and see if there are any other hidden deps you might want to add.